src/osd/OSD.cc

```diff
@@ -3432,10 +3432,25 @@ int OSD::update_crush_location()
 
 int OSD::update_crush_device_class()
 {
+  if (!cct->_conf->osd_class_update_on_start) {
+    dout(10) << __func__ << " osd_class_update_on_start = false" << dendl;
+    return 0;
+  }
+
+  if (store->get_type() != "bluestore") {
```

Comment:
Instead of making this bluestore specific, how about
<pre>
string device_class = store->is_rotational() ? "hdd" : "ssd";
</pre>